cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -36,6 +36,77 @@ namespace modmesh
 
 namespace detail
 {
+
+#if defined(__aarch64__)
+#define vec_typ(typ, N) typ##x##N##_t
```

Comment:
I would recommend to not use `typ`. Instead spelling out will be better `_type`.